hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/test/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapterTest.java

```diff
@@ -226,4 +227,23 @@ public void testIncompatibleDeletesAreDropped() {
     verify(metricsExporter, times(1)).incCounters(INCOMPATIBLE_MUTATION_METRIC_KEY, 1);
     verify(metricsExporter, times(1)).incCounters(DROPPED_INCOMPATIBLE_MUTATION_METRIC_KEY, 1);
   }
+
+  @Test
+  public void testTimestampsOverflowMutations() {
+    ArrayList<Cell> walEntryCells = new ArrayList<>();
+    Cell put1 = new KeyValue(rowKey, cf, qual, Long.MAX_VALUE, KeyValue.Type.Put, val);
+    Cell put2 = new KeyValue(rowKey, cf, qual, 0, KeyValue.Type.Put, val);
+
+    walEntryCells.add(put1);
+    walEntryCells.add(put2);
+    BigtableWALEntry walEntry =
+        new BigtableWALEntry(System.currentTimeMillis(), walEntryCells, tableName);
+
+    Assert.assertEquals(
+        Arrays.asList(put1, put2), incompatibleMutationAdapter.adaptIncompatibleMutations(walEntry));
+
+    verify(metricsExporter).incCounters(INCOMPATIBLE_MUTATION_METRIC_KEY, 0);
```

Comment:
Why should we not increment INCOMPATIBLE_MUTATION_COUNT metric? This is an incompatible mutation, right?

Reply:
I thought of creating a different metric INCOMPATIBLE_MUTATION_TIMESTAMP_OVERFLOW_METRIC_KEY for flagging this. 

Should INCOMPATIBLE_MUTATION_TIMESTAMP_OVERFLOW_METRIC_KEY and INCOMPATIBLE_MUTATION_COUNT be exclusive?

Reply:
They don't need to be exclusive. If we go this route, it maybe helpful to then log metrics for all the incompatible mutation types.

Reply:
@jhambleton  comments?

Reply:
I think this might be helpful for customers if we log metrics for each incompatible type.